template/interface_cpp.njk

```diff
@@ -38,12 +38,26 @@ void {{name}}Bridge::Init(Napi::Env env, Napi::Object exports) {
 
 {{name}}Bridge::{{name}}Bridge(const Napi::CallbackInfo& info)
     : Napi::ObjectWrap<{{name}}Bridge>(info) {
-  // TODO: implement constructor overloading
-  {% for idl_constructor in constructors %} new {{name}} (
-    {% for argument in idl_constructor.arguments %}
-      NativeTypeTraits<IDL{{argument.type | camelcase-}}>::NativeValue(info.Env(), info[{{loop.index - 1}}]){% if not loop.last %},{% endif %}
-    {% endfor %})
-  {% endfor %};
+  if ({{constructors.length}} == 0 && info.Length() == 0) {
+    new {{name}} ();
+    return;
+  }
+
+  {% for idl_constructor in constructors %} 
+    if (info.Length() == {{idl_constructor.arguments.length}}
+      {% for argument in idl_constructor.arguments %} 
+        && NativeTypeTraits<IDL{{argument.type | camelcase-}}>::IsTypeEquals(info[{{loop.index - 1}}])
```

Comment:
I think it's better to put this line in separated ```if``` statement instead of this ```&&``` operation.
If the arguments's lengths are the same, there is no reason to check the length again I think.

Reply:
Okay I will change it!